reward/staking_manager.go

```diff
@@ -242,6 +246,101 @@ func GetStakingInfoOnStakingBlock(stakingBlockNumber uint64) *StakingInfo {
 	return calcStakingInfo
 }
 
+// PreloadStakingInfo preloads staking info for the given headers.
+// It first finds the first block that does not have state, and then
+// it regenerates the state from the nearest block that has state to the target block to preload staking info.
+// Note that the state is saved every 128 blocks to disk in full node.
+func PreloadStakingInfo(headers []*types.Header) ([]uint64, error) {
```

Comment:
Should headers be sorted in the ascending order of the block number? If so, please mention it in the comment

Reply:
Yes, it should be sorted. I'll add it later. Thanks.